core/models/officeblock.py

```diff
@@ -0,0 +1,15 @@
+from django.db import models
+
+
+class OfficeBlock(models.Model):
+    name = models.CharField(max_length=50,
+                            blank=False, null=False, unique=True)
+
+    def clean(self):
+        self.name = " ".join(self.name.title().split())
```

Comment:
For this to work you'll need to override `.save` and call `self.full_clean()`

Reply:
Thanks, @egichuri  We had missed this.
Done.